code/parse/sexp.cpp

```diff
@@ -4085,6 +4339,12 @@ int build_sexp_string(SCP_string &accumulator, int cur_node, int level, int mode
 			stuff_sexp_text_string(buf, node, mode);
 			accumulator += buf;
 
+		} else if (Sexp_nodes[node].type & SEXP_FLAG_CONTAINER) {
```

Comment:
I haven't dug into the `build_sexp_string()` and `build_extended_sexp_string()` functions and thus have no idea whether this fragment and the similar one below it are correct.